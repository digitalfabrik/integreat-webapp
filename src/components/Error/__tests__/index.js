import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { history } from 'store'

import Error from '..'

jest.mock('react-i18next')
jest.mock('store', () => ({history: {goBack: jest.fn()}}))

describe('Error', () => {
  test('should match snapshot', () => {
    const component = renderer.create(
      <Error error="Error Message"/>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  test('call history when onBack is called', () => {
    const preventDefault = jest.fn()
    const wrapper = shallow(<Error error="Error Message"/>).dive()
    wrapper.instance().goBack({preventDefault})

    expect(history.goBack.mock.calls).toHaveLength(1)
    expect(preventDefault.mock.calls).toHaveLength(1)
  })
})
