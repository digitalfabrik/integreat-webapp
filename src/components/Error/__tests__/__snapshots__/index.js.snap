// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match snapshot 1`] = `
<div>
  <div
    className="centerText"
  >
    errors
  </div>
  <div
    className="centerText"
  >
    <span
      aria-hidden={true}
      className="fa fa-frown-o fa-5x"
    />
  </div>
  <a
    className="centerText"
    href="/"
    onClick={[Function]}
  >
    Go back
  </a>
</div>
`;
